### PR TITLE
Allow customizing integration test resource lifecycle

### PR DESCRIPTION
**Pull request by dimas-b**
* External Storage start/stop commands
* Stargate Container JVM args

### CODE REVIEW

File: persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java [hunk 1]
```diff
@@ -61,6 +64,17 @@ public ExternalStorage() {
     super(ClusterSpec.class, STORE_KEY, Namespace.GLOBAL);
   }
 
+  private Cluster createCluster(ClusterSpec spec, ExtensionContext context) {
```
Comment (tomekl007) [inline persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java:67]: could you show an example of how this new customization will be used from the caller's perspective?

Comment (mpenick) [reply persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java:67]: Doesn't `CcmCluster` provide an example?

File: testing/src/test/java/io/stargate/it/storage/StargateContainer.java [hunk 2]
```diff
@@ -88,7 +91,7 @@
   private static final Queue<Integer> jmxPorts = initJmxPorts(MAX_NODES);
 
   // the first 10 addresses are reserved for storage nodes
-  private static final AtomicInteger stargateAddressStart = new AtomicInteger(11);
+  private static final AtomicInteger stargateAddressStart = new AtomicInteger(101);
```
[the remark this answers is not in the document]
Comment (mpenick) [reply testing/src/test/java/io/stargate/it/storage/StargateContainer.java:94]: If there's concern about IP clashes we might consider using a different subnet e.g. `127.0.2.X`? (`127.0.1.X` is already used for the proxy protocol stuff)

[the remark this answers is not in the document]
Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/StargateContainer.java:94]: yes, lower IP numbers are used by various docker images

[the remark this answers is not in the document]
Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/StargateContainer.java:94]: `127.0.2.X` looks interesting. Shall I change to this IP range?

[the remark this answers is not in the document]
Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/StargateContainer.java]: Updated to `127.0.2.X` - thanks for the suggestion!

**Review by mpenick — APPROVED** [review]
Looks good!

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**